llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -7266,12 +7266,18 @@ static SDValue LowerFunnelShift(SDValue Op, SelectionDAG &DAG) {
     MVT VT = Op.getSimpleValueType();
 
     if (Op.getOpcode() == ISD::FSHL) {
+      if (ShiftNo->isZero())
+        return Op.getOperand(0);
```

Comment:
I wasn't sure whether to repeat the combine or just bypass the rewrite to `ISD::FSHR` and let the final DAG combiner run do its thing.  However, this way means we can be sure the result can be isel'd, is about the same code size change and allows for a better unit test.